src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -5621,6 +5621,18 @@ private VariableDeclaratorSyntax ParseVariableDeclarator(
                     break;
 
                 default:
+                    Debug.Assert(argumentList is null);
```

Comment:
Should we also assert that `initializer is null`?

Reply:
Yup.  Done.